frontend/app-development/features/appPublish/components/Deploy/DeployDropdown/DeployDropdown.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { type ReactElement } from 'react';
+import classes from './DeployDropdown.module.css';
+import { StudioCombobox, StudioError, StudioSpinner } from '@studio/components';
+import type { ImageOption } from '../../ImageOption';
+import { useTranslation } from 'react-i18next';
+import { useAppReleasesQuery } from 'app-development/hooks/queries';
+import { BuildResult } from 'app-shared/types/Build';
+import { DateUtils } from '@studio/pure-functions';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { DeployPopover } from './DeployPopover';
+
+export type DeployDropdownProps = {
+  appDeployedVersion: string;
+  disabled: boolean;
+  setSelectedImageTag: (tag: string) => void;
+  selectedImageTag: string;
+  startDeploy: () => void;
+  isPending: boolean;
+};
+
+export const DeployDropdown = ({
+  appDeployedVersion,
+  selectedImageTag,
+  setSelectedImageTag,
+  disabled,
+  startDeploy,
+  isPending,
+}: DeployDropdownProps): ReactElement => {
+  const { org, app } = useStudioEnvironmentParams();
+  const { t } = useTranslation();
+
+  const {
+    data: releases = [],
+    isPending: releasesIsPending,
+    isError: releasesIsError,
+  } = useAppReleasesQuery(org, app, { hideDefaultError: true });
+
+  if (releasesIsPending)
+    return (
+      <StudioSpinner showSpinnerTitle={false} spinnerTitle={t('app_deployment.releases_loading')} />
+    );
+
+  if (releasesIsError) return <StudioError>{t('app_deployment.releases_error')}</StudioError>;
+
+  const imageOptions: ImageOption[] = releases
+    .filter((image) => image.build.result === BuildResult.succeeded)
+    .map((image) => ({
+      value: image.tagName,
+      label: t('app_deployment.version_label', {
+        tagName: image.tagName,
+        createdDateTime: DateUtils.formatDateTime(image.created),
+      }),
+    }));
+
+  return (
+    <div className={classes.deployDropDown}>
+      <StudioCombobox
+        size='small'
+        value={selectedImageTag && imageOptions?.length > 0 ? [selectedImageTag] : undefined}
```

Comment:
Consider extracting this logic into a well-named variable to improve readability and reduce inline complexity. For example:

```ts
const hasSelectedImageTag =  selectedImageTag && imageOptions?.length > 0;
const selectedVersion= hasSelectedImageTag ? [selectedImageTag] : undefined;  
```

```suggestion
        value={selectedVersion}
```

Reply:
Smart! 😄